Kernel/FileSystem/VirtualFileSystem.cpp

```diff
@@ -710,10 +710,8 @@ KResult VFS::symlink(StringView target, StringView linkpath, Custody& base)
         return inode_or_error.error();
     auto& inode = inode_or_error.value();
     auto target_buffer = UserOrKernelBuffer::for_kernel_buffer(const_cast<u8*>((const u8*)target.characters_without_null_termination()));
-    ssize_t nwritten = inode->write_bytes(0, target.length(), target_buffer, nullptr);
-    if (nwritten < 0)
-        return KResult((ErrnoCode)-nwritten);
-    return KSuccess;
+    auto result = inode->write_bytes(0, target.length(), target_buffer, nullptr);
+    return result.result();
```

Comment:
This looks a little iffy. If `write_bytes` succeeds, the returned `KResultOr<ssize_t>` from there would be in the "Or" state, rather than the "KResult" state. So calling `result()` feels wrong..

Reply:
Yea, that's fair. Fixed.